src/serde.rs

```diff
@@ -0,0 +1,149 @@
+use crate::qdrant::value::Kind;
+use crate::qdrant::{ListValue, Struct, Value};
+use serde::ser::{SerializeMap, SerializeSeq};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::collections::HashMap;
+
+impl Serialize for Value {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if let Some(value) = &self.kind {
+            match value {
+                Kind::NullValue(_v) => serializer.serialize_none(),
+                Kind::DoubleValue(double) => serializer.serialize_f64(*double),
+                Kind::IntegerValue(int) => serializer.serialize_i64(*int),
+                Kind::StringValue(str) => serializer.serialize_str(str),
```

Comment:
maybe prefer `r#str` instead of `str` for variables?

same for `r#bool`

Reply:
I am not familiar with this naming convention, what is the rational behind it?

Reply:
It's a way of escaping them to avoid conflicts with the rust keywords, but it's not giving any problems as it is rn

Reply:
These aren't reserved [keywords](https://doc.rust-lang.org/reference/keywords.html), and won't become that in the current Rust edition. It may shadow types but I guess that is fine.

We can always change it to `r#` if it becomes an issue in new editions.